### DISCUSSION TITLE
Enable SSO for directus behind nginx

### DISCUSSION THREAD

**Discussion by younky-yang** [directus/directus#7790] [Archive]
Just setup a directus instance and has nginx plays the reverse proxy role that enables https connections for directus, but after I set the SSO by setting these AUTH_ env variables, the SSO failed as its connection to Azure AD's access url. 

How can I solve the problem by enable SSO for directus but without losing the https connection?

**Comment by rijkvanzanten** [comment]
What's the configuration for PUBLIC_URL and your SSO setup?

**Reply by younky-yang** [reply]
I know the reason now, directus use node https module to send the request to access_url, but it doesn't support proxy settings through https_proxy. which I got the timeout issue when sending request to the access_url. 

In node_modules/request-compose/request/send.js file. 
we may use global-agent package to solve this.

Update:
Add global-agent package and import it from api/src/start.ts by adding the following line.
import 'global-agent/bootstrap';

**Comment by younky-yang** [comment]
https://github.com/directus/directus/pull/8223

**Reply by joselcvarela** [reply]
The solution by @younky-yang was to use 'global-agent' a package that can alter http requests to use proxy configurations.

Our recommended way to solve is to run node and pass 'global-agent' as required argument:
```
$ export GLOBAL_AGENT_HTTP_PROXY=http://127.0.0.1:8080
$ node -r 'global-agent/bootstrap' node_modules/directus/dist/start.js
```

Their documentation: https://github.com/gajus/global-agent#setup-proxy-using-global-agentbootstrap

**Reply by Fx4-X** [reply]
how to achieve same in docker ?

I believe we need to edit this line for node command, but in the docker its npx , little confused. any help ?
 https://github.com/directus/directus/blob/a230a7c2cf4cd7e66cbd7cd84f2099f9b5ed3560/docker/Dockerfile#L63

**Reply by rijkvanzanten** [reply]
@Fx4-X instead of `npx directus start`, use `node -r 'global-agent/bootstrap' node_modules/directus/dist/start.js` in your Dockerfile 👍🏻

**Reply by joselcvarela** [reply]
You can just use node too. There's no difference.
So your command could be ` CMD npx directus bootstrap && export GLOBAL_AGENT_HTTP_PROXY=http://127.0.0.1:8080 && $ node -r 'global-agent/bootstrap' node_modules/directus/dist/start.js`

Don't forget to change the `GLOBAL_AGENT_HTTP_PROXY` 😉

**Reply by Fx4-X** [reply]
Thanks @rijkvanzanten , @joselcvarela , it did work

**Comment by burka** [comment]
It would be great if this information could be added to https://docs.directus.io/self-hosted/sso.html as HTTP_PROXY is not working for SSO as expected.
